test/controllers/sections/firstTransporter/FirstTransporterVatControllerSpec.scala

```diff
@@ -71,41 +72,32 @@ class FirstTransporterVatControllerSpec extends SpecBase with MockUserAnswersSer
     }
 
     "must populate the view correctly on a GET when the question has previously been answered" in new Fixture(
-      Some(emptyUserAnswers.set(FirstTransporterVatPage, "answer"))) {
+      Some(emptyUserAnswers.set(FirstTransporterVatPage, FirstTransporterVatModel(true, Some("Answer"))))) {
 
       val result = testController.onPageLoad(testErn, testDraftId, NormalMode)(request)
 
       status(result) mustEqual OK
-      contentAsString(result) mustEqual view(form.fill("answer"), firstTransporterVatSubmitAction)(dataRequest(request), messages(request)).toString
+      contentAsString(result) mustEqual view(form.fill(FirstTransporterVatModel(true, Some("Answer"))), firstTransporterVatSubmitAction)(dataRequest(request), messages(request)).toString
     }
 
     "must redirect to the next page when valid data is submitted" in new Fixture() {
       MockUserAnswersService.set().returns(Future.successful(emptyUserAnswers))
 
-      val req = FakeRequest(POST, firstTransporterVatRoute).withFormUrlEncodedBody(("value", "answer"))
+      val req = FakeRequest(POST, firstTransporterVatRoute).withFormUrlEncodedBody(("hasVat", "true"), ("firstTransporterVatNumber", testVatNumber))
```

Comment:
Nothing major but this may be more readable:
```suggestion
      val req = FakeRequest(POST, firstTransporterVatRoute).withFormUrlEncodedBody("hasVat" -> "true", "firstTransporterVatNumber" -> testVatNumber)
```